main.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  # my_each()
+  def my_each()
+    each do |item|
+      yield item if block_given?
+    end
+  end
+  # ending my_each()
+
+  # my_each_with_index()
+  def my_each_with_index()
+    index = 0
+    each do |item|
+      yield(item, index) if block_given?
+      index += 1
+    end
+  end
+  # my_each_with_index()
+
+  # my_select
+  def my_select()
+    new_arr = []
+    each do |item|
+      new_arr << item if yield(item)
+    end
+    new_arr
+  end
+  # my_select
+
+  # my_all?
+  def my_all?()
+    flag = true
+    each do |item|
+      feedback = yield(item)
+      flag = false if (feedback == false) || feedback.nil?
+    end
+    flag
+  end
```

Comment:
- [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil

- [ ] my_all? when no block or argument is given returns false when one of the collection members are false or nil

- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer

- [ ] enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric

- [ ] enumerables #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class

- [ ] enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex

- [ ] my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex

- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

- [ ] my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern